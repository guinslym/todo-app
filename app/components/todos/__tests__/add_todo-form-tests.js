import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import AddTodoForm from '../add-todo-form';

const INPUT_VALUE = 'Testing';
const onTodoSubmit = jasmine.createSpy('onTodoSubmit');

describe('<AddTodoForm/>', () => {
  let component;

  beforeEach(() => component = mount(<AddTodoForm onTodoSubmit={onTodoSubmit}/>));

  it('should render', () => {
    expect(component.length).toBeTruthy();
  });

  it('should set the value of the input to empty initially', () => {
    expect(component.find('input').getDOMNode().value).toEqual('');
  });

  describe('When submitting the form', () => {

    beforeAll(() => {
      component.find('input').getDOMNode().value = INPUT_VALUE;
      component.find('form').simulate('submit', { preventDefault: () => {} });
    });

    it('should pass a callback to the onClick handler with value of input', () => {
      expect(onTodoSubmit).toHaveBeenCalled();
    });

    it('should reset the value of the input back to empty', () => {
      expect(component.find('input').getDOMNode().value).toEqual('');
    });

  });

});

describe('<AddTodoForm/> snapshot', () => {

  it('should render correctly', () => {
    const tree = renderer.create(<AddTodoForm onTodoSubmit={onTodoSubmit}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
