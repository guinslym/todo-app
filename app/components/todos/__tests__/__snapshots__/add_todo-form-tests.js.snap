// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodoForm/> snapshot should render correctly 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input />
    <button
      type="submit"
    >
      Add todo
    </button>
  </form>
</section>
`;
